src/rez/package_cache.py

```diff
@@ -466,6 +466,15 @@ def add_variants_async(self, variants):
                     stderr=out_target,
                     **kwargs
                 )
+
+                if (platform.system() == "Windows"
+                        and sys.version_info < (3, 7)):
+                    # Wait subprocess cleanup
+                    #   This is a Python<=3.6 won't fix on Windows.
+                    #   Based on issue: https://bugs.python.org/issue37380
+                    #   and the comment: https://bugs.python.org/msg346332
+                    _.wait(1)
```

Comment:
won't this raise a TimeoutExpired exception if the rez-pkg-cache daemon runs for longer than 1 sec? (Or am I missing something - this seems highly probable).

Reply:
Sorry, somehow I got really sloppy on this one, and you are right. It raises `TimeoutExpired` exception when the process runs longer than 1 sec.

I have changed to use Python 3.7 and didn't revisit this PR since then.
Now that I am back and I found the reason of this issue seems to be related to garbage collect...

Changing `_.wait(1)` into `import gc;gc.collect()` solved the problem. (but we shouldn't need this...)

One additional info, I got the idea that the issue might be related to the garbage collection is from this [comment](https://bugs.python.org/issue37380#msg348231), and here's the [repro](https://bugs.python.org/file48493/invalid_handle.py) that's been mentioned.

Reply:
Oh, jsut do `del _` before function return or not keeping the reference of the subprocess instance in variable `_` solved the problem. :O